pkgs/applications/misc/houdini/default.nix

```diff
@@ -32,6 +55,10 @@ buildFHSEnv rec {
   '';
 
   runScript = writeScript "${name}-wrapper" ''
-    exec $@
+    if [ -d /run/opengl-driver/etc/OpenCL/vendors ]; then
+      exec ${bubblewrap}/bin/bwrap --dev-bind / / --bind /run/opengl-driver/etc/OpenCL/vendors $(realpath /etc/OpenCL)/vendors $@
+    else
+      exec $@
+    fi
```

Comment:
I don't a nested bubblewrap is a good solution here.

Also, if bwrap works here, why use buildFHSEnvChroot?

Reply:
bwrap here is only needed to bind OpenCL/vendors to the ones in running system. 

I just did not find another way to do it, since it cannot be done during build. Do you have suggestions?

So with `buildFHSEnvChroot` and this one bwrap there are no nested bwraps, right?

as for why `buildFHSEnv` does not work - i seem to have figured it out. the reason is that it has `--die-with-parent` option passed to bwrap, and houdini loves to spawn processes in bg. Even by default it forks and exits shortly after start, which leads to bwrap parent process exiting and sending SIGKILL to houdini. Also it spawns hserver process that is meant to be accessed by multiple houdini instances, but having `--unshare-pid` option makes houdini fail to find it if starting several instances in parallel, which leads to licensing errors.... Long story short - i think it's doable to switch to `buildFHSEnv`, but i'd stick with the Chroot one for now, as it behaves in a more expected way.

Reply:
nice! I've only looked into manual and haven't seen these extra arguments (still pretty new to nix in general)

switched to `buildFHSEnv` now, everything seem to work fine

also doing a `--ro-bind-try /etc/OpenCL/vendors /etc/OpenCL/vendors` to address non-NixOS case.  
Assuming both `/etc/OpenCL/vendors` and  `/run/opengl-driver/etc/OpenCL/vendors/` won't ever be present on a system at the same time